src/main.ts

```diff
@@ -68,245 +60,81 @@ export async function activate(
         throw error;
     }
 
-    const aiKey = context.extension.packageJSON.contributes.debuggers[0].aiKey;
-    const reporter = new TelemetryReporter(aiKey);
-    // ensure it gets properly disposed. Upon disposal the events will be flushed.
-    context.subscriptions.push(reporter);
+    // Verify that the current platform is supported by the extension and inform the user if not.
+    if (!checkIsSupportedPlatform(context, platformInfo)) {
+        return null;
+    }
 
-    const csharpchannelObserver = new CsharpChannelObserver(csharpChannel);
-    const csharpLogObserver = new CsharpLoggerObserver(csharpChannel);
-    eventStream.subscribe(csharpchannelObserver.post);
-    eventStream.subscribe(csharpLogObserver.post);
+    await checkDotNetRuntimeExtensionVersion(context);
 
-    const requiredPackageIds: string[] = ['Debugger'];
+    await MigrateOptions(vscode);
+    const optionStream = createOptionStream(vscode);
 
-    requiredPackageIds.push('Razor');
+    const requiredPackageIds: string[] = ['Debugger', 'Razor'];
 
     const csharpDevkitExtension = getCSharpDevKit();
     const useOmnisharpServer = !csharpDevkitExtension && commonOptions.useOmnisharpServer;
     if (useOmnisharpServer) {
         requiredPackageIds.push('OmniSharp');
     }
 
-    const dotnetRuntimeExtensionId = 'ms-dotnettools.vscode-dotnet-runtime';
-    const requiredDotnetRuntimeExtensionVersion = '2.2.3';
-
-    const dotnetRuntimeExtension = vscode.extensions.getExtension(dotnetRuntimeExtensionId);
-    const dotnetRuntimeExtensionVersion = dotnetRuntimeExtension?.packageJSON.version;
-    if (lt(dotnetRuntimeExtensionVersion, requiredDotnetRuntimeExtensionVersion)) {
-        const button = vscode.l10n.t('Update and reload');
-        const prompt = vscode.l10n.t(
-            'The {0} extension requires at least {1} of the .NET Install Tool ({2}) extension. Please update to continue',
-            context.extension.packageJSON.displayName,
-            requiredDotnetRuntimeExtensionVersion,
-            dotnetRuntimeExtensionId
-        );
-        const selection = await vscode.window.showErrorMessage(prompt, button);
-        if (selection === button) {
-            await vscode.commands.executeCommand('workbench.extensions.installExtension', dotnetRuntimeExtensionId);
-            await vscode.commands.executeCommand('workbench.action.reloadWindow');
-        } else {
-            throw new Error(
-                vscode.l10n.t(
-                    'Version {0} of the .NET Install Tool ({1}) was not found, {2} will not activate.',
-                    requiredDotnetRuntimeExtensionVersion,
-                    dotnetRuntimeExtensionId,
-                    context.extension.packageJSON.displayName
-                )
-            );
-        }
-    }
-
-    // If the dotnet bundle is installed, this will ensure the dotnet CLI is on the path.
-    await initializeDotnetPath();
```

Comment:
this seemed to be useless.  Just waits on the extension pack activation (we don't have a dependency on it) and then asks for the sdk acquisition status, but does nothing with it.

Confirmed with Noah that asking for SDK acquisition status won't do anything like install an SDK